File: tests/fixtures/python/PatternMatchSwitch.py
from __future__ import annotations


class PatternMatchSwitch:

    def describe(self, obj: object) -> str:

        def _j2py_switch_1(_j2py_subject: object) -> str:
            match _j2py_subject:
                case int() as i:
                    return f"int: {i}"
                case str() as s if not s:
                    return "empty string"
                case str() as s:
                    return f"string: {s}"
                case None:
                    return "null"
                case _:
                    return f"other: {obj}"
        result = _j2py_switch_1(obj)
        return result

File: tests/fixtures/java/PatternMatchSwitch.java
package com.example;

public class PatternMatchSwitch {
    public String describe(Object obj) {
        String result = switch (obj) {
            case Integer i -> "int: " + i;
            case String s when s.isEmpty() -> "empty string";
            case String s -> "string: " + s;
            case null -> "null";
            default -> "other: " + obj;
        };
        return result;
    }
}
